src/Compilers/CSharp/Test/Symbol/Compilation/UsedAssembliesTests.cs

```diff
@@ -0,0 +1,2727 @@
+    {
+        _ = E1.F1 + 1;
+    }
+}
+");
+
+            verify(comp0Ref, comp1Ref,
+@"
+using static C1<C0>.E1;
+public class C2
+{
+    public static void Main()
+    {
+        _ = F1 + 1;
+    }
+}
+");
+
+            verify(comp0Ref, comp1Ref,
+@"
+using alias = C1<C0>.E1;
+public class C2
+{
+    public static void Main()
+    {
+        _ = alias.F1 + 1;
+    }
+}
+");
+
+            verify(comp0Ref, comp1Ref,
+@"
+using alias = C1<C0>;
+public class C2
+{
+    public static void Main()
+    {
+        _ = alias.E1.F1 + 1;
+    }
+}
+");
+
+            verify(comp0Ref, comp1Ref,
+@"
+public class C2
+{
+    public static void Main()
+    {
+        _ = nameof(C1<C0>.E1.F1);
+    }
+}
+");
+
+            verify(comp0Ref, comp1Ref,
+@"
+using static C1<C0>;
+public class C2
+{
+    public static void Main()
+    {
+        _ = nameof(E1.F1);
+    }
+}
+");
+
+            verify(comp0Ref, comp1Ref,
+@"
+using static C1<C0>.E1;
+public class C2
+{
+    public static void Main()
+    {
+        _ = nameof(F1);
+    }
+}
+");
+
+            verify(comp0Ref, comp1Ref,
+@"
+using alias = C1<C0>.E1;
+public class C2
+{
+    public static void Main()
+    {
+        _ = nameof(alias.F1);
+    }
+}
+");
+
+            verify(comp0Ref, comp1Ref,
+@"
+using alias = C1<C0>;
+public class C2
+{
+    public static void Main()
+    {
+        _ = nameof(alias.E1.F1);
+    }
+}
+");
+
+            verify(comp0Ref, comp1Ref,
+@"
+public class C2
+{
+    public static void Main()
+    {
+        _ = nameof(C1<C0>.C3.F3);
+    }
+}
+");
+
+            verify(comp0Ref, comp1Ref,
+@"
+using static C1<C0>;
+public class C2
+{
+    public static void Main()
+    {
+        _ = nameof(C3.F3);
+    }
+}
+");
+
+            verify(comp0Ref, comp1Ref,
+@"
+using alias = C1<C0>.C3;
+public class C2
+{
+    public static void Main()
+    {
+        _ = nameof(alias.F3);
+    }
+}
+");
+
+            verify(comp0Ref, comp1Ref,
+@"
+using alias = C1<C0>;
+public class C2
+{
+    public static void Main()
+    {
+        _ = nameof(alias.C3.F3);
+    }
+}
+");
+
+            static void verify(MetadataReference reference0, MetadataReference reference1, string source)
+            {
+                AssertUsedAssemblyReferences(source, reference0, reference1);
+            }
+        }
+
+        [Fact]
+        public void FieldReference_06()
+        {
+            var source0 =
+@"
+public class C0 {}
+";
+            var comp0 = CreateCompilation(source0);
+            var comp0Ref = comp0.ToMetadataReference();
+
+            var source1 =
+@"
+public class C1<T>
+{
+    public enum E1
+    {
+        F1 = 0
+    }
+
+    public class C3
+    {
+        public int F3;
+    }
+}
+";
+            var comp1 = CreateCompilation(source1);
+            comp1.VerifyDiagnostics();
+            var comp1Ref = comp1.ToMetadataReference();
+
+            verify(comp0Ref, comp1Ref,
+@"
+class C2
+{
+    /// <summary>
+    /// <see cref=""C1{C0}.E1.F1""/>
```

Comment:
>C0 [](start = 23, length = 2)

I'm not understanding why this wouldn't count as a reference to comp0 when parsing documentation?

Reply:
> I'm not understanding why this wouldn't count as a reference to comp0 when parsing documentation?

Because it is treated as a definition of a type parameter, that could be referred to later. You don't see cref parsing working this way?

---
In reply to: [347630679](https://github.com/dotnet/roslyn/pull/39807#discussion_r347630679) [](ancestors = 347630679)